src/main/java/com/samoilenko/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -17,10 +23,16 @@
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
+    private final EntityManager entityManager;
 
     @Override
+    @Transactional
     public BookDto createBook(BookRequestDto requestDto) {
-        Book savedBook = bookMapper.toModel(requestDto);
+        Book savedBook = bookMapper.toEntity(requestDto);
+        Set<Category> categories = requestDto.getCategoryIds().stream()
+                .map(id -> entityManager.getReference(Category.class, id))
+                .collect(Collectors.toSet());
+        savedBook.setCategories(categories);
```

Comment:
You make N DB requests to save categories, please investigate more efficient way

Reply:
Is way that we discussed on QnA is not working?

Reply:
Working, i will add it in next commit